### DISCUSSION TITLE
ETIMEDOUT: error in every 10-15 days

### DISCUSSION THREAD

**Discussion by deepakkumar4747** [rabbitmq/rabbitmq-server#10900] [Questions]
In every 10-15 days of interval I am getting this error of connection timeout while consuming messages via worker
what is more doubtful is that I am using same connection code while sending message and consuming messgae
but I only gets this error while consuming message and due to which my worker stops consuming and my queue gets stuck.
`Error while connecting to RabbitMQ Error: connect ETIMEDOUT 43.204.11.217:5671
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1300:16) {
  errno: -110,
  code: 'ETIMEDOUT',
  syscall: 'connect',
  address: '43.204.11.217',
  port: 5671
}
TypeError: Cannot read properties of undefined (reading 'assertExchange')
    at RabbitMQTopicHelper.consumeMsg (file:///usr/src/app/src/Helpers/RabbitMQHelper.js:28:26)
    at file:///usr/src/app/src/worker.js:20:12
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)`

**Comment by michaelklishin** [comment]
```
/usr/src/app/src/Helpers/RabbitMQHelper.js:28:26
```

is not a part of RabbitMQ and you haven't shared its contents. We do not guess in this community.

Start with inspecting [server logs](https://www.rabbitmq.com/docs/logging#logged-events) for clues. Chances are your client loses connection and does not recover correctly. How that is done is client library-dependent.

Some other potentially relevant doc guides:

 * [Heartbeats](https://www.rabbitmq.com/docs/heartbeats)
 * [Troubleshooting Networking](https://www.rabbitmq.com/docs/troubleshooting-networking)
 * [Troubleshooting TLS](https://www.rabbitmq.com/docs/troubleshooting-ssl)
